wandb/sdk/launch/builder/build.py

```diff
@@ -139,9 +139,45 @@ def get_docker_user(launch_project: LaunchProject, runner_type: str) -> Tuple[st
     {user} || echo ""
 """
 
-SAGEMAKER_ENTRYPOINT_TEMPLATE = """
-COPY ./src/train {workdir}
-ENTRYPOINT ["sh", "train"]
+ENTRYPOINT_TEMPLATE = """
+COPY ./src/{default_entrypoint} {workdir}
+ENTRYPOINT ["python", "{default_entrypoint}", "{entrypoint}"]
+"""
+
+PYTHON_ENTRYPOINT = """
+import subprocess
+import os
+import sys
+
+def _run_entry_point(command: str):
+    work_dir = os.getcwd()
+    env = os.environ.copy()
+    if os.name == "nt":
+        # we are running on windows
+        process = subprocess.Popen(
+            ["cmd", "/c", command], close_fds=True, cwd=work_dir, env=env
+        )
+    else:
+        process = subprocess.Popen(
+            ["bash", "-c", command],
```

Comment:
Might not want to assume we have bash.  "sh" would be safer to assume.  Doing this in python is a little scary, I'm not sure signals and various other linux stuff will propagate properly.  Since this is just turning env vars into a command that's run it might be simpler to just make this a simple shell script.

Since we're running in docker it seems like we'll never hit the `os.name == "nt"` case.  You can't run windows in docker...

Reply:
Like I think the following would work

```shell
#!/bin/sh

if [[ -z "${WANDB_ENTRYPOINT_COMMAND}" ]]; then
    CMD="$1"
else
    CMD="${WANDB_ENTRYPOINT_COMMAND}"
fi
CMD+=" ${WANDB_ARGS}"
eval $CMD
```

Reply:
Awesome, made some small modifications, and abandonned the python script. Thanks!